src/units/unit.cpp

```diff
@@ -1395,12 +1395,98 @@ bool unit::has_ability_by_id(const std::string& ability) const
 	return false;
 }
 
-void unit::remove_ability_by_id(const std::string& ability)
+static bool matches_if_present_in_filter(const config& filter, const config& cfg, const std::string& attribute, bool def)
+{
+	if(filter[attribute].empty()) {
+		return true;
+	}
+
+	return filter[attribute].to_bool() == cfg[attribute].to_bool(def);
+}
+
+static bool string_matches_if_present_in_filter(const config& filter, const config& cfg, const std::string& attribute, const std::string& def, bool in_tag=false)
```

Comment:
I still haven't looked for existing implementations of this function, but I'd prefer to split this into two functions that would be generic enough to be moved to a general "filter helpers" library. Although that would mean putting an `if(in_tag) {...} else {}` block in `matches_ability_filter`, I think that be worthwhile if we can reuse the generic functions elsewhere.

* One with the current name, that's equivalent to this with `in_tag=false`.
* A `string_list_matches_if_present_in_filter`, which is equivalent to this with `in_tag=true`.

Alternatively, rename `in_tag` to something that would suit a generic function, such as `split_filter_as_list`.

Reply:
done